examples/IrisClassification/iris_classification.py

```diff
@@ -122,4 +125,15 @@ def test_step(self, batch, batch_idx):
     trainer.fit(model, dm)
     trainer.test()
 
-    torch.save(model.state_dict(), "iris.pt")
+
+    # Save model as state dict
+    if os.path.exists("model_state_dict"):
+        shutil.rmtree("model_state_dict")
+    mlflow.pytorch.save_state_dict(pytorch_model=model, path="model_state_dict")
+
+    # Save the entire model
+    if os.path.exists("entire_model"):
```

Comment:
@shrinath-suresh Thanks for the PR, I wonder in case of using ddp as the accelerator, saving the entire model might not be needed,  as if user pass the checkpoints saved in this directory that would fail. However, state_dict would cover for both ddp and non-ddp cases.

Reply:
@shrinath-suresh Also, the model path for creating deployment in the README might need to be synced with the changes.

Reply:
@HamidShojanazeri Thanks Hamid for the comments. This is just a sample example to save and load. You are right, we need to comment out the part of entire_model when running in ddp environment. Its work in progress. I missed to leave a note here in GH.

Reply:
@HamidShojanazeri This has a dependent PR - https://github.com/chauhang/mlflow/pull/33 . We were trying to save the model as state dict and then try the model deployment in CPU/GPU and DDP environment.